### PR TITLE
Nightmare: Add typeInterval constructor option and screenshot overload for clipping path

### PR DESCRIPTION
**Pull request by delesseps**
* screenshot overload - https://github.com/segmentio/nightmare#screenshotpath-clip
* typeInterval - https://github.com/segmentio/nightmare#typeinterval-default-100ms

### CODE REVIEW

File: types/nightmare/index.d.ts [hunk 1]
```diff
@@ -96,6 +96,7 @@ declare class Nightmare {
     removeListener(event: 'error', cb: (msg: string, trace?: Nightmare.IStackTrace[]) => void): Nightmare;
     removeListener(event: 'timeout', cb: (msg: string) => void): Nightmare;
     screenshot(path: string): Nightmare;
+    screenshot(path: string, clip: Object): Nightmare;
```
Comment (DanielRosenwasser) [inline types/nightmare/index.d.ts]: Could you add an explicit type for the shape of `clip`? Either a declared one, or a type literal like `{ x: number, y: ... }`

Also, [according to the source](https://github.com/segmentio/nightmare/blob/a835558067c0dd72b5b10c14613d9071aef39d13/lib/actions.js#L586-L602), it looks like the appropriate overloads here would be

```ts
screenshot(path: string, done?: (err: any) => void): Nightmare;
screenshot(clip: { x: number, y: number, width: number, height: number }, done?: (err: any) => void): Nightmare;
screenshot(path: string, clip?: { x: number, y: number, width: number, height: number }, done: (err: any) => void): Nightmare;
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/nightmare/index.d.ts (hunk 1 above)
Comment (delesseps) [inline types/nightmare/index.d.ts]: Thanks @DanielRosenwasser for the feedback.  I have adjusted and added overloads to handle the buffer being returned too.

**Review by delesseps — COMMENTED** [review]
Have adjusted as per feedback.  Thanks @DanielRosenwasser

**Comment by delesseps** [comment]
Hi @DanielRosenwasser is there anything that needs to be done to get this merged?

**Pull request merged**